pkg/addons/default/coredns.go

```diff
@@ -167,54 +162,81 @@ func InstallCoreDNS(rawClient kubernetes.RawClientInterface, region string, wait
 	return false, nil
 }
 
-// UpdateCoreDNSImageTag updates image tag for kube-system:deployment/coredns based to match the latest release
-func UpdateCoreDNSImageTag(clientSet k8s.Interface, plan bool) (bool, error) {
-	printer := printers.NewJSONPrinter()
+// UpdateCoreDNS will update the `coredns` add-on
+func UpdateCoreDNS(rawClient kubernetes.RawClientInterface, region, controlPlaneVersion string, plan bool) (bool, error) {
```

Comment:
There is a bunch of logic here now, it would be good to test that it is doing the right thing, not only not throwing an error.